File: BenchmarkMemory.py
#benchmark_memory.py
import sys
import json

def measure_memory_usage():
    lst = [i for i in range(1000000)]
    memory_used = sys.getsizeof(lst)
    return memory_used

if __name__ == "__main__":
    result = {"memory_used": measure_memory_usage()}
    print(json.dumps(result))

File: BenchmarkMemory.rb
# benchmark_memory.rb
require 'json'

def measure_memory_usage
  initial_memory = GC.stat(:heap_allocated_pages)
  lst = (0...1000000).to_a
  final_memory = GC.stat(:heap_allocated_pages)
  memory_used = final_memory - initial_memory
  {memory_used: memory_used * GC::INTERNAL_CONSTANTS[:RVALUE_SIZE]}
end

puts measure_memory_usage.to_json
